### ISSUE TITLE
Shoot state sync error for shoots in garden namespace

### ISSUE THREAD

**Issue by timuthy** [gardener/gardener#1989]
**What happened**:

Log from arbitrary Gardenlet:

```
time="2020-02-28T14:31:12Z" level=error msg="Couldn't retrieve Shoot az from namespace garden-garden. Extension Infrastructure state with name az and purpose <nil> was NOT synced: shoots.core.gardener.cloud \"az\" not found" Seed=soil-gcp
```

The reconciler expects the shoot to be in the `garden-garden` namespace but it's in the `garden` namespace. Project related namespaces cannot generally be restored from the shoot namespace in the seed like done here: https://github.com/gardener/gardener/blob/3377a71c400dd2844b3e88404b4e6981de9c5fdf/pkg/gardenlet/controller/federatedseed/shootstate/common_sync.go#L136

/cc @swilen-iwanow @plkokanov

**Environment**:

- Gardener version: v1.0.4, v1.10.0-dev

**Comment by timuthy** [comment]
@plkokanov I skimmed over #1972 but I think it's not enough to add an exception for the `garden` namespace only (https://github.com/gardener/gardener/pull/1972/files#diff-fc78dcc65e3db85a8d3057f877ac7d9cR145) since projcets can use any other namespace defined in `.spec.namespace`

**Comment by plkokanov** [comment]
@timuthy thanks for looking into it, I'll adapt the PR

**Comment by rfranzke** [comment]
@plkokanov can this be closed now that #1972 is merged? Or was it unrelated?

**Comment by rfranzke** [comment]
ping @plkokanov

**Comment by rfranzke** [comment]
I'm closing this issue now as it should be fixed with #1972, kindly reopen if you still see problems.
